### PR TITLE
Update dependencies for build from source, throttle action feedback, add stub for audio command

### LINKED ISSUE #159: Update irobot_create_msgs tag to version with fix for `rosidl_default_generators` dependency
**Describe the bug**
Currently irobot_create_msgs dependency in `dependencies.repos` points to tag 1.2.2 which is missing a build dependency on `rosidl_default_generators`.  This can cause a from scratch sandbox build to fail.  There is expected to be a new tag to support audio shortly.  Even though the simulator will not support audio control, we should update to the next tag to get the build fix (and add a stub subscription for audio).

**To Reproduce**
Checkout fresh ROS 2 sandbox to build all code form source including simulator dependencies.  irobot_create_msgs will fail to build because `rosidl_default_generators` isn't guaranteed to be built before it.

**Expected behavior**
package.xml should trigger `rosidl_default_generators` to be built first

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by justinIRBT**
## Description

- Updated package dependencies based on missing packages that weren't complained about while building from source from scratch
- Throttle Feedback on wall follow / docking / LedAnimation actions
- Add stub subscription for audio command topic `/cmd_audio` to mimic robot.  This won't do anything on simulator other than print a message 

Fixes #159 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tried building source from scratch and was able to get reported build failures after missing packages allowed simulator to build, but fail to run on first iteration.  Build failures require packages to be present before trying to run.  Verified simulator ran without new errors and subscribed to audio topic printed for othe code.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: irobot_create_ignition/irobot_create_ignition_toolbox/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,6 @@ set(dependencies
   control_msgs
   geometry_msgs
   sensor_msgs
-  nav_msgs
```
Comment (justinIRBT) [inline irobot_create_ignition/irobot_create_ignition_toolbox/CMakeLists.txt:47]: This was listed in the list twice

File: irobot_create_common/irobot_create_nodes/include/irobot_create_nodes/mock_publisher.hpp [hunk 2]
```diff
@@ -29,6 +30,7 @@ class MockPublisher : public rclcpp::Node
 
   // Callback functions
   void lightring_callback(irobot_create_msgs::msg::LightringLeds::SharedPtr msg);
+  void audio_callback(irobot_create_msgs::msg::AudioNoteVector::SharedPtr msg);
```
Comment (alsora) [inline irobot_create_common/irobot_create_nodes/include/irobot_create_nodes/mock_publisher.hpp:33]: I created a ticket to rework this mock publisher as it's not really a publisher.
https://github.com/iRobotEducation/create3_sim/issues/162

File: irobot_create_gazebo/irobot_create_gazebo_bringup/package.xml [hunk 3]
```diff
@@ -17,7 +17,9 @@
   <exec_depend>irobot_create_control</exec_depend>
   <exec_depend>irobot_create_description</exec_depend>
   <exec_depend>irobot_create_gazebo_plugins</exec_depend>
+  <exec_depend>joint_state_publisher</exec_depend>
   <exec_depend>ros2launch</exec_depend>
+  <exec_depend>xacro</exec_depend>
```
Comment (alsora) [inline irobot_create_gazebo/irobot_create_gazebo_bringup/package.xml:22]: These added dependencies are not correct.
I don't see them being used in this project, but rather they are only used by `irobot_create_common_bringup` that already lists them

File: irobot_create_ignition/irobot_create_ignition_toolbox/package.xml [hunk 4]
```diff
@@ -14,10 +14,14 @@
   <depend>irobot_create_toolbox</depend>
   <depend>rclcpp</depend>
   <depend>rclcpp_action</depend>
+  <depend>rclpy</depend>
```
Comment (alsora) [inline irobot_create_ignition/irobot_create_ignition_toolbox/package.xml:17]: This is needed due to this line https://github.com/iRobotEducation/create3_sim/blob/main/irobot_create_ignition/irobot_create_ignition_toolbox/CMakeLists.txt#L28
But the error is in the CMakeLists.txt as it's not true that this project depends on `rclpy.
We should remove the dependency from there

File: irobot_create_ignition/irobot_create_ignition_toolbox/package.xml [hunk 5]
```diff
@@ -14,10 +14,14 @@
...
   <depend>rcutils</depend>
   <depend>ros_ign_interfaces</depend>
   <depend>std_msgs</depend>
   <depend>sensor_msgs</depend>
+  <depend>nav_msgs</depend>
```
Comment (alsora) [inline irobot_create_ignition/irobot_create_ignition_toolbox/package.xml:22]: nitpick: move this above to maintain alphabetical order

File: irobot_create_gazebo/irobot_create_gazebo_bringup/package.xml (hunk 3 above)
Comment (justinIRBT) [reply irobot_create_gazebo/irobot_create_gazebo_bringup/package.xml:22]: Hmmm... I was able to build and run all packages for the gazebo sim, but when I ran the simulator, those 2 packages weren't in my sandbox and it was a runtime failure.  Why did it build irobot_create_common_bringup if those packages were on my system?

**Review by alsora — APPROVED** [review]

**Review by alsora — APPROVED** [review]

**Pull request merged**